### PR TITLE
fix for #2309 : Forked JVM executing a task using ProActive hangs

### PR DESCRIPTION
**Pull request by fviale**
- Added a System.exit(0) call at the end of the forkedJVM main method
- Added a test which reproduces the problem

### CODE REVIEW

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java [hunk 1]
```diff
@@ -34,17 +34,12 @@
...
-
+import org.apache.commons.io.FileUtils;
 import org.ow2.proactive.scheduler.task.TaskResultImpl;
 import org.ow2.proactive.scheduler.task.context.TaskContext;
 import org.ow2.proactive.scheduler.task.executors.InProcessTaskExecutor;
-import org.apache.commons.io.FileUtils;
+
+import java.io.*;
 
 public class ExecuteForkedTaskInsideNewJvm {
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1118)

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java [hunk 2]
```diff
@@ -77,6 +72,9 @@ public static void main(String[] args) throws Throwable {
         }
 
         fromForkedJVM(args[0]);
+
+        // Call to System.exit is necessary at this point (when the task is finished normally) as the forked JVM can keep alive non-daemon threads
+        System.exit(0);
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java]: CRITICAL Remove this call to "exit" or ensure it is really required. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1147)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 15 issues:
* CRITICAL 5 critical
* MAJOR 8 major
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Catch Exception instead of Throwable.](https://github.com/ow2-proactive/scheduling/blob/9d5b6c0d532f162368474e2ca8caf96cf500e272/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L87) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1181)
* MAJOR [Replace this usage of System.out or System.err by a logger.](https://github.com/ow2-proactive/scheduling/blob/9d5b6c0d532f162368474e2ca8caf96cf500e272/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L84) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)
* MAJOR [Replace this usage of System.out or System.err by a logger.](https://github.com/ow2-proactive/scheduling/blob/9d5b6c0d532f162368474e2ca8caf96cf500e272/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L84) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)
* MINOR [Catch Exception instead of Throwable.](https://github.com/ow2-proactive/scheduling/blob/9d5b6c0d532f162368474e2ca8caf96cf500e272/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L91) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1181)
* MAJOR [Replace this usage of System.out or System.err by a logger.](https://github.com/ow2-proactive/scheduling/blob/9d5b6c0d532f162368474e2ca8caf96cf500e272/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L93) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)
* CRITICAL [Use a logger to log this exception.](https://github.com/ow2-proactive/scheduling/blob/9d5b6c0d532f162368474e2ca8caf96cf500e272/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L93) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1148)
* CRITICAL [Use a logger to log this exception.](https://github.com/ow2-proactive/scheduling/blob/9d5b6c0d532f162368474e2ca8caf96cf500e272/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L88) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1148)
* MAJOR [Replace this usage of System.out or System.err by a logger.](https://github.com/ow2-proactive/scheduling/blob/9d5b6c0d532f162368474e2ca8caf96cf500e272/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L70) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)
* MAJOR [Replace this usage of System.out or System.err by a logger.](https://github.com/ow2-proactive/scheduling/blob/9d5b6c0d532f162368474e2ca8caf96cf500e272/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L88) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)
* CRITICAL [Remove this call to "exit" or ensure it is really required.](https://github.com/ow2-proactive/scheduling/blob/9d5b6c0d532f162368474e2ca8caf96cf500e272/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L71) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1147)
* CRITICAL [Remove this call to "exit" or ensure it is really required.](https://github.com/ow2-proactive/scheduling/blob/9d5b6c0d532f162368474e2ca8caf96cf500e272/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L95) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1147)
* MAJOR [Replace this usage of System.out or System.err by a logger.](https://github.com/ow2-proactive/scheduling/blob/9d5b6c0d532f162368474e2ca8caf96cf500e272/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L92) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)
* MAJOR [Either log or rethrow this exception.](https://github.com/ow2-proactive/scheduling/blob/9d5b6c0d532f162368474e2ca8caf96cf500e272/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L91) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

**Comment by lpellegr** [comment]
:+1:

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java (hunk 2 above)
Comment (fviale) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java]: yes it is :-)

**Pull request merged**